ssl-config/src/main/java/com/palantir/ssl/SslSocketFactories.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.ssl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import com.palantir.ssl.SslConfiguration.Builder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * Utility functions for creating {@link SSLSocketFactory}s
+ * that are configured with Java trust stores and key stores.
+ */
+public final class SslSocketFactories {
+    private SslSocketFactories() {}
+
+    /**
+     * Create an {@link SSLSocketFactory} using the provided configuration.
+     *
+     * @param trustStorePath  location of the trust store
+     * @param trustStoreType  optional type of the trust store, defaults to {@code JKS} when not provided
+     * @param trustStorePassword optional password for the trust store (generally not required)
+     * @return an {@link SSLSocketFactory} according to the input configuration
+     */
+    public static SSLSocketFactory createSslSocketFactory(
+            Path trustStorePath,
+            Optional<String> trustStoreType,
+            Optional<String> trustStorePassword) {
+        Builder builder = SslConfiguration.builder();
+        builder.trustStorePath(trustStorePath);
+        builder.trustStoreType(trustStoreType);
+        builder.trustStorePassword(trustStorePassword);
+
+        return createSslSocketFactory(builder.build());
+    }
+
+    /**
+     * Create an {@link SSLSocketFactory} from the provided configuration.
+     *
+     * @param config a {@link SslConfiguration} describing the location, type and password of the desired trust
+     *               store.
+     * @return an {@link SSLSocketFactory} according to the input configuration
+     */
+    public static SSLSocketFactory createSslSocketFactory(SslConfiguration config) {
+        SSLContext sslContext = createSslContext(config);
+        return sslContext.getSocketFactory();
+    }
+
+    /**
+     * Create an {@link SSLContext} initialized from the provided configuration.
+     */
+    @VisibleForTesting
```

Comment:
This is needed for line 201 of `SslSocketFactoriesConnectionTests`:

```SSLServerSocketFactory factory = sslContext.getServerSocketFactory();```

In the connection tests we want to create a *server* socket factory using the context, but the ```createSslSocketFactory``` returns a regular (non-server) socket factory.

Since the focus of this library seems to be for clients, I wasn't sure that exposing a separate ```createSslServerSocketFactory``` made sense.